File: src/main/java/name/zeno/zervice/logic/GoodsLogic.java
package name.zeno.zervice.logic;

import name.zeno.zervice.dao.GoodsDao;
import name.zeno.zervice.dao.entity.Goods;
import org.nutz.dao.pager.Pager;
import org.nutz.ioc.loader.annotation.Inject;
import org.nutz.ioc.loader.annotation.IocBean;

import java.util.List;

@IocBean
public class GoodsLogic
{
    @Inject
    private GoodsDao goodsDao;

    public List<Goods> query(Pager pager)
    {
        return goodsDao.query();
    }

    public Goods create(Goods goods)
    {
        return goodsDao.insert(goods);
    }

    public Goods update(Goods goods)
    {
        return goodsDao.update(goods);
    }

    public int delete(long goodsId)
    {
        return goodsDao.delete(goodsId);
    }
}

File: src/main/java/name/zeno/zervice/logic/GoodsLogic.kt
package name.zeno.zervice.logic

import name.zeno.zervice.dao.GoodsDao
import name.zeno.zervice.dao.entity.Goods
import org.nutz.dao.pager.Pager
import org.nutz.ioc.loader.annotation.Inject
import org.nutz.ioc.loader.annotation.IocBean

@IocBean
class GoodsLogic {
  @Inject
  lateinit private var goodsDao: GoodsDao

  fun query(pager: Pager?=null): List<Goods> = goodsDao.query()
  fun create(goods: Goods): Goods = goodsDao.insert(goods)
  fun update(goods: Goods): Goods = goodsDao.update(goods)
  fun delete(goodsId: Long): Int = goodsDao.delete(goodsId)
}
